jwst/flatfield/flat_field.py

```diff
@@ -73,7 +74,8 @@ def do_correction(input_model,
     # NIRSpec spectrographic data are processed differently from other
     # types of data (including NIRSpec imaging).  The test on flat is
     # needed because NIRSpec imaging data are processed by do_flat_field().
-    if input_model.meta.instrument.name == 'NIRSPEC' and flat is None:
+    if input_model.meta.instrument.name == 'NIRSPEC' and \
```

Comment:
It's not obvious to me that you need to check both instrument name and exp_type here, since it would be pretty impossible to have instrument name be anything but NIRSPEC if the exp_type is in that list, but of course it can't hurt anything either.

Reply:
Yes, that is probably right. Will remove the instrument check. Just being too conservative on the changes.